README.md

```diff
@@ -50,61 +58,100 @@ time, in order to decide the exact way in which play must be executed. Take for
             .map(previousTwoGames -> {
               // ... do something with the two most recent game state representations
             }));
 ```
 
-### Agents, Rules, and Formations
+### Identities, Rules, and Formations
 
-A game (usually) consists of at least one game agent. This game agent can interact with and mutate
-the actual state of the game. In the example above the `Pawn` is a data representation of such an
-agent. `zosma` requires that each agent representation be identifiable, ie. it should be able to
+A game (usually) contains some objects which can interact with each other. These interactions mutate
+state of the game. In the example below the `ChessPiece` is a data representation of such an
+object. `zosma` requires that each of these objects be identifiable, i.e. it should be able to
 supply an identity (getters omitted for brevity):
 
 ```java
-public class Pawn implements Identity.Supplier {
-  private final Location location;
-  private final Identity identity;
-  private final Color    teamColor;
+@Value
+public class ChessPiece implements Identity.Supplier {
+  private final Location            location;
+  private final ChessPiece.Identity identity;
+
+  @Value
+  public static class Identity implements io.leonis.zosma.game.Identity {
+    private final ChessPiece.Type type;
+    private final ChessGame.Player    owner;
+  }
+  
+  public enum Type {
+    KING,
+    QUEEN,
+    ROOK,
+    BISHOP,
+    KNIGHT,
+    PAWN
+  }
 }
 ```
 
-Usually a strategy will specify some desired configuration of agents. This configuration can be
+A strategy will often specify some desired configuration of agents. This configuration can be
 purely positional, but can also be used for assigning roles to agents (not applicable to chess, 
 since chess roles are not dynamic). For example:
 
 ```java
 public class ChessFormation implements Formation<Pawn, Location> {
-  private final Map<Pawn, Location> desiredFormation;
+  private final Map<ChessPiece, Location> desiredFormation;
   
   @Override
-  public Location getFormationFor(final Pawn pawn) {
+  public Location getFormationFor(final ChessPiece piece) {
     return this.desiredFormation.get(pawn);
   }
 }
 ```
 
+The legality of an action in a game and the win-conditions are determined by the rules of the game. 
+The game agent must know these rules in order to determine the validity of its own potential moves
+or to anticipate a specific state of the game. A rule is evaluated by examining a game state:
+
+```java
+public class LoseCondition implements Rule<ChessGame, ChessGame.Player> {
+  Set<ChessGame.Player> getViolators(final ChessGame input) {
+    // a player has lost, if
+    return input.getPawnPositions().keySet().stream()
+      // its king
+      .filter(chessPiece -> chessPiece.getIdentity().getType().equals(ChessPiece.Type.KING))
+      // is off the board
+      .filter(chessPiece -> chessPiece.getLocation.equals(Location.OUTSIDE)
```

Comment:
![image](https://user-images.githubusercontent.com/4689114/33828292-cd03680c-de6b-11e7-9478-c660f256ba24.png)

Reply:
You could statically import Location and ChessPiece.Type and just reference the enum's directly. It'd make the code more concise.